parcels/particlesets/particlesetsoa.py

```diff
@@ -45,9 +47,20 @@ def _to_write_particles(pd, time):
             & (np.isfinite(pd['time'])))
 
 
-class ParticleAccessor(object):
+class ParticleAccessorSOA(BaseParticleAccessor):
     def __init__(self, pset):
         self.pset = pset
+        self.max_len = pset.size
+        self._index = 0
+        self._iterindex = 0
```

Comment:
Working on this I begin to see a bit better why the Accessor and Iterator were implemented as separate objects and I am no longer convinced that it is definitely best to merge the two.

Reply:
hm, okay - let's perhaps talk about that either Friday or Wednesday - looking forward to your explanation!